person.rb

```diff
@@ -0,0 +1,21 @@
+class Person
+  attr_accessor :name, :age
+  attr_reader :id
+
+  def initialize(age, name: 'Unknown', parent_permission: true)
+    @id = nil
```

Comment:
- [ ] [OPTIONAL] Instead of initializing id as nil, I would suggest initializing it with a unique identifier like SecureRandom.uuid. This way, you can avoid potential conflicts or collisions with other objects that might have the same id. Also, you can use keyword arguments for name and parent_permission, so that you can pass them in any order when creating a new object. For example, instead of this:

```ruby

def initialize(age, name: 'Unknown', parent_permission: true)
  @id = nil
  @name = name
  @age = age
  @parent_permission = parent_permission
end

```

do this:

```rb

def initialize(age, name: 'Unknown', parent_permission: true)
  @id = SecureRandom.uuid
  @age = age
  @name = name
  @parent_permission = parent_permission
end

```

Reply:
Thank you so much for this corrections